src/ingestors/foundation/index.ts

```diff
@@ -0,0 +1,133 @@
+
+const CONTRACT_ADDRESS = '0x62037b26ffF91929655AA3A060F327b47d1e2b3e';
+
+const getChainId = (chain: string) => {
+  switch (chain) {
+    case 'base':
+      return 8453;
+    case 'eth':
+    default:
+      return 1;
+  }
+};
+
+export class FoundationIngestor implements MintIngestor {
+  async supportsUrl(_resources: MintIngestorResources, url: string): Promise<boolean> {
+    return new URL(url).hostname === 'www.foundation.app' || new URL(url).hostname === 'foundation.app';
+  }
+
+  async supportsContract(resources: MintIngestorResources, contractOptions: MintContractOptions): Promise<boolean> {
+    if (contractOptions.chainId !== 8453) {
+      return false;
+    }
+    const contract = await getFoundationMintByAddress(resources, contractOptions);
+    if (!contract) {
+      return false;
+    }
+    return true;
+  }
+
+  async createMintForContract(
+    resources: MintIngestorResources,
+    contractOptions: MintContractOptions,
+  ): Promise<MintTemplate> {
+    const mintBuilder = new MintTemplateBuilder()
+      .setMintInstructionType(MintInstructionType.EVM_MINT)
+      .setPartnerName('foundation');
+
+    if (contractOptions.url) {
+      mintBuilder.setMarketingUrl(contractOptions.url);
+    }
+
+    const collection = await getFoundationMintByAddress(resources, contractOptions);
+
+    if (!collection) {
+      throw new MintIngestorError(MintIngestionErrorName.CouldNotResolveMint, 'Collection not found');
+    }
+
+    const contractAddress = collection.contractAddress;
+
+    const description = collection?.description;
+    const image = collection?.media.url;
+
+    mintBuilder.setName(collection.name).setDescription(description).setFeaturedImageUrl(image);
+    mintBuilder.setMintOutputContract({ chainId: 8453, address: contractAddress });
+    mintBuilder.addImage(collection.banner_image_url, 'banner');
+
+    if (!collection.creator) {
+      throw new Error('Error finding creator');
+    }
+
+    mintBuilder.setCreator({
+      name: collection.creator.name,
+      walletAddress: collection.creator.publicKey,
+    });
+
+    const totalPriceWei = await getFoundationMintPriceInWei(
+      8453,
+      CONTRACT_ADDRESS,
+      contractAddress,
+      resources.alchemy,
+      collection.saleType,
+    );
+
+    // Proof is always this
+    const proof = '["0x00000000000000000000000000000000000000000000000000000000000000a0", "0x0000000000000000000000000000000000000000000000000000000000000000"]';
+
+    mintBuilder.setMintInstructions({
+      chainId: 8453,
+      contractAddress: CONTRACT_ADDRESS,
+      contractMethod:
+        collection.saleType === 'FIXED_PRICE_DROP'
+          ? 'mintFromFixedPriceSaleWithEarlyAccessAllowlistV2'
+          : 'mintFromDutchAuctionV2',
+      contractParams:
+        collection.saleType === 'FIXED_PRICE_DROP'
+          ? `["${collection.contractAddress}", 1, "${contractOptions.recipient}", "0x0000000000000000000000000000000000000000", ${proof}]`
```

Comment:
This is generating an `undefined` value for recipient. Can you make sure the generated contractParams work?

In this case, I think you're looking for the macro `address`, as used here:
https://github.com/floornfts/mobile-minting/blob/main/src/ingestors/fxhash/index.ts#L67

This will be expanded to be the recipient's address